src/main/java/com/nutomic/syncthingandroid/syncthing/SyncthingService.java

```diff
@@ -415,6 +492,13 @@ public void onConfigChanged() {
             });
 
             mEventProcessor = new EventProcessor(SyncthingService.this, mApi);
+            mEventProcessor.addOnEventBasedModelChangedListener(new EventBasedModel.OnEventBasedModelChangedListener() {
+                @Override
+                public void onEventBasedModelChanged(EventBasedModel model) {
+                    // TODO: needs to run on main thread!
```

Comment:
You can create a `Handler` on the main thread, then do `handler.post()`. Or maybe there's already a handler in this class.